From c157947a3b3bd0c9cba8542c92d24281ebfb2ea7 Mon Sep 17 00:00:00 2001
From: inogenous <123803852+inogenous@users.noreply.github.com>
Date: Tue, 12 Sep 2023 20:30:23 +0200
Subject: [PATCH] Bugfix: AIM: Display correct truncation of container names

* Prevent string truncation inside `<color>` tags of container names in AIM
* For example, instead of displaying `plastic bottle (sealed) > <color` before, this commit will display `plastic bottle (sealed) > clean water (frozen...`
* Truncation to AIM width is still performed by `trim_and_print` anyway, so there should be little need to also use `utf8_truncate`.
---
 src/advanced_inv.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/advanced_inv.cpp b/src/advanced_inv.cpp
index a1ec73312b9aa..6aa1ad893eafd 100644
--- a/src/advanced_inv.cpp
+++ b/src/advanced_inv.cpp
@@ -795,8 +795,7 @@ void advanced_inventory::redraw_pane( side p )
     const advanced_inv_area &sq = same_as_dragged ? squares[AIM_DRAGGED] : square;
     bool car = square.can_store_in_vehicle() && panes[p].in_vehicle() && sq.id != AIM_DRAGGED;
     std::string name = utf8_truncate( car ? sq.veh->name : sq.name, width );
-    std::string desc = utf8_truncate( pane.container ? pane.container->tname( 1, false ) :
-                                      sq.desc[car], width );
+    std::string desc = pane.container ? pane.container->tname( 1, false ) : sq.desc[car];
     // starts at offset 2, plus space between the header and the text
     width -= 2 + 1;
     trim_and_print( w, point( 2, 1 ), width, active ? c_green  : c_light_gray, name );
